### ISSUE TITLE
Wrapping fake throws TargetInvocationException instead of the original exception

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#836]
The following code:
```c#
void Main()
{
    var foo = new Foo();
    var wrapper = A.Fake<IFoo>(o => o.Wrapping(foo));
    wrapper.Bar(null);
}

public interface IFoo
{
    int Bar(string s);
}

public class Foo : IFoo
{
    public int Bar(string s)
    {
        return s.Length;
    }
}
```
throws a `TargetInvocationException`, but calling `foo.Bar(null)` directly throws a `NullReferenceException`. The wrapping fake should throw the original exception.

**Comment by blairconrad** [comment]
Good one!

**Comment by blairconrad** [comment]
This issue has been released in [FakeItEasy 2.3.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.3.0).
